### PR TITLE
Document the way the WHATWG operates

### PR DESCRIPTION
**Pull request by annevk**

### CODE REVIEW

**Comment by annevk** [comment]
I put a preview up on https://html5.org/temp/working-mode.html.

Feedback I have thus far:

 - [ ] Add IDs.
 - [ ] Make other pages link here.

**Comment by annevk** [comment]
Another thing I haven't looked at is how to revise the current FAQ. I'm not sure that needs to be a blocker, but that's obviously something we'd have to work on as well.

File: src/working-mode [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+   <li><a href="/" rel="home">Home</a></li>
+   <li><a href="https://spec.whatwg.org/">Standards</a></li>
+   <li class="this"><strong>Working Mode</strong></li>
+  </ul>
+
+  <h2>Working Mode</h2>
+
+  <p>The WHATWG develops standards and assists in maintaining their corresponding tests. The WHATWG
+  advances the web by creating standards and tests, and thereby fostering independent
+  implementations. The WHATWG adheres to a shared code of conduct.
```
Comment (domenic) [inline src/working-mode]: Code of conduct should link

File: src/working-mode [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+
+  <p>It’s not always possible to get the answers to all these questions in a timely manner.
+  Therefore, it might be the case that a decision is made even without input from all parties. This
+  makes it very important to file bugs for any spec changes (as discussed below), as a final check
+  to make sure all implementers are on board with, or at least aware of, the change.
+
+  <p>Additions to the standard can also be proposed as issues. However, the process for turning such
+  an issue into an accepted change to the standard is generally more involved, as the criteria for
+  additions is stricter (see Additions). A typical path for turning such issues into accepted
+  changes is detailed in the New Proposals section.
```
Comment (domenic) [inline src/working-mode]: Additions and New Proposals should link to the corresponding sections

File: src/working-mode [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+  <ul>
+   <li>The addition must have the support of at least two implementers. (Such support is not binding
+   in any way on the implementers, however.)
+  </ul>
+
+  <p>Additionally, the following are strongly recommended:
+
+  <ul>
+   <li>The support from implementers should be of the form "we would like to implement this soon"
+   and not just "this seems like a reasonable idea".
```
Comment (domenic) [inline src/working-mode]: These quotes are not smart; probably they should be. (Or the other smart quotes should be removed.)

There are several other instances of this later in the document.

File: src/working-mode [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+  often not available, or the shape of a feature is not yet clear enough for implementers to feel
+  comfortable pledging their support.
+
+  <p>In such scenarios we anticipate features will be "incubated" outside of WHATWG standards. This
+  could be in a variety of venues, such as:
+
+  <ul>
+   <li><p>Another standards organization
+   <li><p>A personal document or GitHub repository
+   <li><p>A pull request to a WHATWG Standard, with the understanding it will not be merged until
```
Comment (domenic) [inline src/working-mode]: Lowercase "standard" here

File: src/working-mode [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+
+  <p>As described above, the criteria for inclusion in a WHATWG standard is rather strict. In the
+  initial stages of feature development for the web platform, such widespread implementer support is
+  often not available, or the shape of a feature is not yet clear enough for implementers to feel
+  comfortable pledging their support.
+
+  <p>In such scenarios we anticipate features will be "incubated" outside of WHATWG standards. This
+  could be in a variety of venues, such as:
+
+  <ul>
```
Comment (domenic) [inline src/working-mode]: Nit: other `<ul>`s have newlines between their `<li>`s; this one is missing them.

File: src/working-mode [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+   <li><p>Another standards organization
+   <li><p>A personal document or GitHub repository
+   <li><p>A pull request to a WHATWG Standard, with the understanding it will not be merged until
+   the criteria above are met
+   <li>
+    <p>A document hosted by the WHATWG, but not under the spec.whatwg.org subdomain and not titled
+    "Living Standard"
+
+    <p class="example"><a href="https://wiki.whatwg.org/wiki/CanvasColorSpace">CanvasColorSpace</a>.
```
Comment (domenic) [inline src/working-mode]: I'd expand this to a full sentence, such as "The [CanvasColorSpace] proposal is being incubated on [the WHATWG wiki]."

File: src/working-mode [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+  </ul>
+
+  <p>In all cases, features that hope to graduate to a WHATWG standard should strive to follow the
+  above guidelines, gather appropriate implementer commitments, and have corresponding tests.
+
+  <p>Additionally, those maintaining such proposals should try to involve the WHATWG community, for
+  example by filing an issue on the standard they anticipate eventually becoming a part of.
+
+  <p class="note">This process of incubation can be very lightweight, such as filing a pull request
```
Comment (domenic) [inline src/working-mode]: Maybe prepend "Overall, ..." so it looks less like this is a note about the above paragraph and more about the whole section.

File: src/working-mode [hunk 8]
```diff
@@ -0,0 +1,242 @@
...
+  on the appropriate WHATWG standard with a proposal, and gathering appropriate implementer support
+  there. Or it could be more involved, such as creating a separate repository at another venue and
+  iterating for a long time there.
+
+  <p class="example">The addition of <code>URL.prototype.toJSON()</code> was rather straightforward
+  once a plan was agreed upon and upstream IDL issues were resolved:
+  <a href="https://github.com/whatwg/url/issues/137">whatwg/url#137</a>.
+
+  <p class="example"><code>innerText</code> was initially drafted in a repository outside the HTML
```
Comment (domenic) [inline src/working-mode]: Maybe link "a repository outside the HTML Standard" to https://github.com/rocallahan/innerText-spec/

File: src/working-mode [hunk 9]
```diff
@@ -0,0 +1,242 @@
...
+  Standard. Integration into HTML then went rather smoothly:
+  <a href="https://github.com/whatwg/html/pull/1678">whatwg/html#1678</a>.
+
+  <h3>Tests</h3>
+
+  <p>High-level guidelines for any tests:
```
Comment (domenic) [inline src/working-mode]: I guess we never fleshed this section out. I'd revise the introductory sentence to be something like

> As discussed in [Changes], normative changes to the standard require tests, usually submitted the [web-platform-tests] project. At a high level, tests should strive to:
>
> - Be automated
> - Test error handling
> - Test limits and edge cases, such as overflow outside of the expected value set

File: src/working-mode [hunk 10]
```diff
@@ -0,0 +1,242 @@
...
+
+  <ul>
+   <li>Should be automated.
+   <li>Should test error handling.
+   <li>Should test limits. (E.g., how is overflow handled?)
+  </ul>
+
+  <h3>Conflicts</h3>
+
+  <p>In case of a conflict, the editor decides. However, the editor is expected to go to significant
```
Comment (domenic) [inline src/working-mode]: Maybe "a conflict among the community of contributors"? Or leave as-is.

File: src/working-mode [hunk 11]
```diff
@@ -0,0 +1,242 @@
...
+  length to resolve disagreements.
+
+  <p>Implementations can always override the editor by implementing something else. Whenever that
+  happens a breakdown in communication has taken place that the community should seek to resolve.
```
Comment (domenic) [inline src/working-mode]: "to resolve" sounds a little weird; maybe "to reach a resolution".

File: src/working-mode [hunk 12]
```diff
@@ -0,0 +1,242 @@
...
+
+  <p>Implementations that disagree can be rather tricky to resolve. Generally, we try to resolve
+  these situations as follows:
```
Comment (domenic) [inline src/working-mode]: s/try to resolve/approach to remove some repetition

File: src/working-mode (hunk 5 above)
Comment (domenic) [inline src/working-mode]: I guess this document is rather inconsistent about `<li>` vs. `<li><p>` too, hrm.

Comment (annevk) [inline src/working-mode]: I used the latter only when a list item contained multiple paragraphs.

**Comment by annevk** [comment]
We should probably do a short blog post about this as well.

**Comment by annevk** [comment]
Draft blog post for those who want to review: https://docs.google.com/document/d/1fxQo6sIm0DNPrnIriOu4bgUmQOL8pyAoyzQ7HrXborM/edit?usp=sharing.

**Comment by annevk** [comment]
(I haven't checked the styles in detail either. I wish we had an easier way of creating a test or preview instance.)

File: src/working-mode [hunk 13]
```diff
@@ -0,0 +1,252 @@
...
+  </ul>
+
+  <h2>Working Mode</h2>
+
+  <p>The WHATWG develops standards and assists in maintaining their corresponding tests. The WHATWG
+  advances the web by creating standards and tests, and thereby fostering independent
+  implementations. The WHATWG adheres to a shared
+  <a href="https://wiki.whatwg.org/wiki/Code_of_Conduct">Code of Conduct</a>.
+
+  <p>While discussing and working on standards, it’s always useful to keep these questions in mind:
```
Comment (foolip) [inline src/working-mode]: Since it wasn't consistent, I assume that the use of non-ASCII quotes here and elsewhere was due to copy-paste and will put a commit to make the document all ASCII. (Consistent fancy quotes would also be fine.)

File: src/working-mode [hunk 14]
```diff
@@ -0,0 +1,252 @@
...
+  </h1>
+  <ul class="navigation">
+   <li><a href="/" rel="home">Home</a></li>
+   <li><a href="https://spec.whatwg.org/">Standards</a></li>
+   <li class="this"><strong>Working Mode</strong></li>
+  </ul>
+
+  <h2>Working Mode</h2>
+
+  <p>The WHATWG develops standards and assists in maintaining their corresponding tests. The WHATWG
```
Comment (foolip) [inline src/working-mode]: The repetition here is a bit unfortunate for the opening paragraph. Will try to massage it a bit.

File: src/working-mode [hunk 15]
```diff
@@ -0,0 +1,252 @@
...
+
+   <li>There should be no strong implementer objections to the new feature.
+
+   <li>There should already be a prototype implementation or one being worked on side-by-side with
+   the change to the standard.
+
+   <li>New features should be something that cannot be done today or are such a common trend in
+   libraries that standardizing them helps lots of people. Alternatively, the "new" feature could
+   actually be a legacy feature that was not yet standardized, which is now being standardized. (A
+   legacy "addition" is important to allow for more meaningful competition and interoperability.)
```
Comment (foolip) [inline src/working-mode]: I read this as "important to allow" as if legacy things were not allowed by some default. Trying a small rewrite.

File: src/working-mode (hunk 3 above)
Comment (foolip) [inline src/working-mode]: Didn't get to this comment before I ASCII-ified the document, but I did resolve it in a way.

File: src/working-mode [hunk 16]
```diff
@@ -141,8 +141,9 @@
 
    <li>New features should be something that cannot be done today or are such a common trend in
    libraries that standardizing them helps lots of people. Alternatively, the "new" feature could
-   actually be a legacy feature that was not yet standardized, which is now being standardized. (A
-   legacy "addition" is important to allow for more meaningful competition and interoperability.)
+   actually be a legacy feature that was not yet standardized, which is now being standardized.
+   (Legacy features are often hard to remove due to web compatibility, and stardizing and testing
+   them help improve interoperability.)
```
Comment (domenic) [inline src/working-mode]: s/stardizing/standardizing

s/help/helps

**Comment by annevk** [comment]
I'm a little sad with the loss of fancy quotes, I thought I had made them consistent, but alas, no big deal.

Pushed one further fixup. Quick look appreciated, but will probably put this live in a couple of hours, including the blog post.

**Comment by foolip** [comment]
I'll put the fancy quotes back and fix the ones that weren't.

**Comment by foolip** [comment]
Done, no more tweaking from me.

**Comment by annevk** [comment]
That's very kind, thanks! All good then?

**Comment by foolip** [comment]
Ship it!
